src/native/fs/fs_napi.cpp

```diff
@@ -1846,6 +1848,16 @@ struct FileFcntlLock : public FSWrapWorker<FileWrap>
                 SetError("invalid lock type");
             }
         }
+        if (info.Length() > 2 && !info[2].IsUndefined()) {
+            auto op_string = info[2].As<Napi::String>().Utf8Value();
+            if (op_string == "THROW") {
+                op = F_OFD_SETLK;
```

Comment:
We might want to quickly confirm that this works as expected on GPFS and CephFS. Although most likely it would but just to be on the safe side. We had some surprises with flock behaviour on GPFS initially.

Reply:
I think we will try to avoid surprises and use the functionality you implemented for Glacier, I just updated the PR with the changes, I had to take some of your code to util file, can you please check?
Also as I asked on Slack, I saw that on Glacier, for restore and migrate you check the timestamp file with exceeded_time() and in expiry you check is_desired_time(), can you please elaborate on the differences? I thought that in my case is_desired_time() should fit but I would be glad to get your opinion here